WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,273 @@
+        public var sizeLimit: Int = 1024 * 1024 * 100
+
+        /// When performing a sweep, the cache will remote entries until the size of
+        /// the remaining items is lower than or equal to `sizeLimit * trimRatio` and
+        /// the total count is lower than or equal to `countLimit * trimRatio`. `0.7`
+        /// by default.
+        var trimRatio = 0.7
+
+        /// The number of seconds between each LRU sweep. 30 by default.
+        /// The first sweep is performed right after the cache is initialized.
+        ///
+        /// Sweeps are performed in a background and can be performed in parallel
+        /// with reading.
+        public var sweepInterval: TimeInterval = 30
+
+        /// The delay after which the initial sweep is performed. 10 by default.
+        /// The initial sweep is performed after a delay to avoid competing with
+        /// other subsystems for the resources.
+        var initialSweepDelay: TimeInterval = 10
+
+        /// Initializes the configuration.
+        ///
+        /// - parameter sizeLimit: Size limit in bytes. `100 Mb` by default.
+        public init(sizeLimit: Int = 1024 * 1024 * 100) {
+            self.sizeLimit = sizeLimit
+        }
+    }
+
+    /// Creates a cache instance with a given `name`. The cache creates a directory
+    /// with the given `name` in a `.cachesDirectory` in `.userDomainMask`.
+    public init(name: String, configuration: Configuration = .init()) {
+        let cachesURL = DiskCache.getCachesURL()
+        self.init(url: cachesURL.appendingPathComponent(name, isDirectory: true), configuration: configuration)
+    }
+
+    private static func getCachesURL() -> URL {
+        guard let cachesURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            DDLogError("Failed to instantiate cache. Caches directory not available")
+            return URL(fileURLWithPath: "/dev/null") // This should never happen
+        }
+        return cachesURL
+    }
+
+    /// Creates a cache instance with a given root URL.
+    public init(url: URL, configuration: Configuration = .init()) {
+        self.rootURL = url
+        self.configuration = configuration
+
+        do {
+            try FileManager.default.createDirectory(at: rootURL, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            DDLogError("Failed to creates cache root directory at: \(url) with error: \(error)")
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + configuration.initialSweepDelay) { [weak self] in
+            Task {
+                await self?.performAndScheduleNextSweep()
+            }
+        }
+    }
+
+    // MARK: Codable
+
+    public func getValue<T: Decodable>(
+        _ type: T.Type,
+        forKey key: String,
+        decoder: JSONDecoder = JSONDecoder()
+    ) async -> T? {
+        guard let data = getData(forKey: key) else { return nil }
```

Comment:
Please don't ignore IO errors – they should always be passed up the stack for the caller to deal with.

Reply:
I strongly feel that they should not be passed up the stack in this case of a disk cache, and returning an optional is good.